helix-view/src/document.rs

```diff
@@ -119,6 +121,7 @@ pub struct Document {
     pub(crate) modified_since_accessed: bool,
 
     diagnostics: Vec<Diagnostic>,
+    text_annotations: HashMap<TextAnnotationGroup, Vec<TextAnnotation>>,
```

Comment:
Might be more performant to implement this as a `BTreeMap` in tne future with the lines as keys.
This would make iterating a small range of lines (during rendering) more efficient when there are a large amount of annotations.
Can probably be done in a followup PR tough